### ISSUE TITLE
`02286_quantile_tdigest_infinity` slower with new analyzer because of constant folding

### ISSUE THREAD

**Issue by antonio2368** [ClickHouse/ClickHouse#62231]
It's easy to verify by running the test with enabled and disabled analyzer.

The problematic part is the `arrayResize` and you can simply compare performance of:
```sql
select arrayResize(arrayResize([inf], 500000, -inf), 1000000, inf);
```

I would expect that the function should be execute anyway at some point but it's obviously slower when we do it during constant folding. Also, memory usage is 10x higher.

cc @novikd @KochetovNicolai

**Comment by antonio2368** [comment]
There are multiple reasons:
- transforming constant to string
   - https://github.com/ClickHouse/ClickHouse/blob/master/src/Planner/PlannerActionsVisitor.cpp#L252
   - https://github.com/ClickHouse/ClickHouse/blob/master/src/Analyzer/ConstantNode.cpp#L26
 - creating/destroying Field object

**Comment by antonio2368** [comment]
Easiest option would be to simply disable constant folding for `arrayResize`.

**Comment by novikd** [comment]
@antonio2368 we are aware of the problem with big constants. The actual issue is that we store `Field` instead of `IColumn` in `ConstantNode`. We are working on it #59523.

**Comment by KochetovNicolai** [comment]
Fixed by #62336
